equipment_handler/test/sband_binary_tests.c

```diff
@@ -0,0 +1,131 @@
+//
+//    ret = STX_getControl(&pa_status, &mode);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("pa_status: %d  mode: %d\n", pa_status, mode);
+//
+//    ret = STX_getPaPower(&pa_power);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("pa_power: %d\n", pa_power);
+//
+//    ret = STX_getFrequency(&freq);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("S-band frequency: %f\n", freq);
+//
+//    ret = STX_getBuffer(0, &count);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("Buffer count: %d\n", count);
+//
+//    ret = STX_getBuffer(1, &underrun);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("Buffer underrun: %d\n", underrun);
+//
+//    ret = STX_getBuffer(2, &overrun);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("Buffer overrun: %d\n\n", overrun);
+//
+//
+//    uint8_t message[46] = "Quentinen and Tarantined by Writtin Directino ";
+//    uint16_t data[23] = {0};
+//
+//    ret = STX_setFrequency(2228.0f);
+//    if(ret != S_SUCCESS) return ret;
+//    ret = STX_setPaPower(24u);
+//    if(ret != S_SUCCESS) return ret;
+//    ret = STX_setControl(S_PA_ENABLE, S_SYNC_MODE);
+//    if(ret != S_SUCCESS) return ret;
+//
+//
+//    for(int k = 0; k++; k<23){
+//        data[k] = (message[2*k] << 8) || message[2*k+1];
+//    }
+//
+//    // Send message 200 times over SPI to fill buffer
+//    for(int i = 0; i < 400;i++){
+//        SPISbandTx(data, 23);
+//    }
+//
+//    ret = STX_getBuffer(0, &count);
+//    printf("Buffer count at start of transmission: %d\n", count);
+//    int transmit = 0;
+//    int i = 0;
+//
+//    // Set transmitter to data mode to begin transmission
+//    ret = STX_setControl(S_PA_ENABLE, S_DATA_MODE);
+//    while(!transmit){
+//        ret = STX_getTR(&transmit);
+//        i++;
+//    }
+//
+//    // Set to Synchronization mode, PA on
+//    ret = STX_setControl(S_PA_ENABLE, S_SYNC_MODE);
+//
+//    vTaskDelay(0.5* ONE_SECOND);
+//
+//    printf("The transmit ready flag was checked %d times\n\n", i);
+//
+//    // Set to Configuration mode, PA off
+//    ret = STX_setControl(S_PA_DISABLE, S_CONF_MODE);
+//
+//    printf("Configuration at END of test:\n");
+//
+//    ret = STX_getControl(&pa_status, &mode);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("pa_status: %d  mode: %d\n", pa_status, mode);
+//
+//    ret = STX_getPaPower(&pa_power);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("pa_power: %d\n", pa_power);
+//
+//    ret = STX_getFrequency(&freq);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("S-band frequency: %f\n", freq);
+//
+//    ret = STX_getBuffer(0, &count);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("Buffer count: %d\n", count);
+//
+//    ret = STX_getBuffer(1, &underrun);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("Buffer underrun: %d\n", underrun);
+//
+//    ret = STX_getBuffer(2, &overrun);
+//    if(ret != S_SUCCESS) return ret;
+//    printf("Buffer overrun: %d\n\n", overrun);
```

Comment:
Why is this commented out?

Reply:
It doesn't need to be, but it's commented out as a precaution. We don't want this running inadvertently because it actually transmits from the S-band radio and we don't want to risk damaging anything if a call to sband_binary_test() is left somewhere in the code. Running this should be a very deliberate thing, requiring the user to uncomment the code.

Reply:
Gotcha. Best case scenario would probably be to exclude from production builds in the toolchain parameters to do what you say, but that's not set up.